### PR TITLE
[lockfile-explorer] Add support for PNPM 10

### PR DESCRIPTION
**Pull request by octogonz**
## Summary

📌 This is mostly working, but PNPM lockfiles mix together exact versions with peer dependency SemVer ranges, and Lockfile Explorer apparently has always confused that.  I need to straighten that out.

### CODE REVIEW

File: apps/lockfile-explorer-web/src/packlets/lfx-shared/IJsonLfxGraph.ts [hunk 1]
```diff
@@ -18,13 +18,13 @@ export interface IJsonPeerDependencyMeta {
 
 export interface IJsonLfxDependency {
   name: string;
-  version: string;
+  versionPath: string;
```
Comment (iclanton) [inline apps/lockfile-explorer-web/src/packlets/lfx-shared/IJsonLfxGraph.ts:21]: Maybe leave a doc comment about what this is.

File: apps/lockfile-explorer-web/src/packlets/lfx-shared/LfxGraph.ts [hunk 2]
```diff
@@ -6,39 +6,75 @@ import type { IJsonLfxWorkspace } from './IJsonLfxWorkspace';
 
 export interface ILfxGraphDependencyOptions {
   name: string;
-  version: string;
-  dependencyType: LfxDependencyKind;
-  containingEntry: LfxGraphEntry;
-  peerDependencyMeta: IJsonPeerDependencyMeta;
+  versionPath: string;
```
Comment (iclanton) [inline apps/lockfile-explorer-web/src/packlets/lfx-shared/LfxGraph.ts:9]: Same here.

File: apps/lockfile-explorer/src/graph/test/fixtures/edge-cases/edge-cases.md [hunk 3]
```diff
@@ -2,3 +2,6 @@
 
 This test fixture is a PNPM workspace crafted to reproduce interesting edge cases in the `lfxGraphLoader` algorithm.
 
+The lockfiles were built from this repistory:
+
+https://github.com/octogonz/lockfile-explorer-edge-cases
```
Comment (iclanton) [inline apps/lockfile-explorer/src/graph/test/fixtures/edge-cases/edge-cases.md:7]: Why not just check them into this repo?

File: common/changes/@rushstack/lockfile-explorer/octogonz-lfx-pnpm-10_2025-10-03-05-47.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/lockfile-explorer",
+      "comment": "Add support for PNPM 10 lockfile format",
+      "type": "major"
```
Comment (iclanton) [inline common/changes/@rushstack/lockfile-explorer/octogonz-lfx-pnpm-10_2025-10-03-05-47.json:6]: Why not minor?

**Review by iclanton — APPROVED** [review]

File: apps/lockfile-explorer-web/src/packlets/lfx-shared/IJsonLfxGraph.ts (hunk 1 above)
Comment (octogonz) [reply apps/lockfile-explorer-web/src/packlets/lfx-shared/IJsonLfxGraph.ts:21]: These are just the serialized versions of the corresponding fields from `LfxGraph`

File: apps/lockfile-explorer-web/src/packlets/lfx-shared/LfxGraph.ts (hunk 2 above)
Comment (octogonz) [reply apps/lockfile-explorer-web/src/packlets/lfx-shared/LfxGraph.ts:9]: These all have the same meaning as the corresponding fields in `LfxGraph`, since they are just its inputs

File: common/changes/@rushstack/lockfile-explorer/octogonz-lfx-pnpm-10_2025-10-03-05-47.json (hunk 4 above)
Comment (octogonz) [reply common/changes/@rushstack/lockfile-explorer/octogonz-lfx-pnpm-10_2025-10-03-05-47.json:6]: I think we're too conservative about bumping our versions :)

File: apps/lockfile-explorer/src/graph/test/fixtures/edge-cases/edge-cases.md (hunk 3 above)
Comment (octogonz) [reply apps/lockfile-explorer/src/graph/test/fixtures/edge-cases/edge-cases.md:7]: We should do that. I was deliberating about whether to make it a separate repo like https://github.com/microsoft/rush-example

**Comment by octogonz** [comment]
I'll address these minor points in the next PR
